### PR TITLE
Add TypeScript types for <Overlay>, <OverlayTrigger>, <Tooltip>, <IconButton>

### PR DESCRIPTION
**Pull request by bradenmacdonald**
## Description

This adds types for `<Overlay>`, `<OverlayTrigger>`, `<Tooltip>`, and `<IconButton>` by converting their implementations from `.jsx` to `.tsx`.

This also fixes a number of random type errors that then showed up in the various test files, and makes some minor cleanups to `<Chip>` related code, such as removing `// @ts-ignore` directives now that we have typings available for `<IconButton>`.

### Deploy Preview

https://deploy-preview-3100--paragon-openedx.netlify.app/

## Merge Checklist

n/a for this type of internal change.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@openedx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @bradenmacdonald! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: www/src/components/CodeBlock.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ const {
 
 export type CollapsibleLiveEditorTypes = {
   children: React.ReactNode;
-  clickToCopy: (arg: string) => void;
```
Comment (brian-smith-tcril) [inline www/src/components/CodeBlock.tsx:40]: @adamstankiewicz do you know why `arg: string` was in here before?

**Review by brian-smith-tcril — APPROVED** [review]
This is wonderful! I'm super happy to see components cleaned up like this!

LGTM!

File: www/src/components/CodeBlock.tsx (hunk 1 above)
Comment (PKulkoRaccoonGang) [reply www/src/components/CodeBlock.tsx:40]: Most likely, we previously passed some kind of parameter here, but now this parameter does not exist.
Good catch! 💯

**Review by PKulkoRaccoonGang — COMMENTED** [review]
This looks great, left some suggestions

**Comment by bradenmacdonald** [comment]
@PKulkoRaccoonGang Thanks for the review! I made those suggested changes.

**Comment by bradenmacdonald** [comment]
@brian-smith-tcril or @PKulkoRaccoonGang could one of you please merge this at your convenience? :)

**Review by PKulkoRaccoonGang — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@bradenmacdonald 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/3100&entry.1484655318=https://github.com/bradenmacdonald&entry.752974735=2024-06-17+18:46&entry.1917517419=2024-07-07+11:20&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by PKulkoRaccoonGang** [comment]
@bradenmacdonald 🚀

**Pull request merged**